controller/array_action/array_mediator_svc.py

```diff
@@ -407,9 +407,11 @@ def _copy_to_target_volume(self, target_volume_name, source_volume_name):
             self._rollback_copy_to_target_volume(target_volume_name)
             raise ex
 
-    def copy_to_existing_volume_from_source(self, name, source_name, source_capacity_in_bytes,
-                                            minimum_volume_size_in_bytes, pool=None):
-        self._copy_to_target_volume(name, source_name)
+    def copy_to_existing_volume_from_source(self, volume_id, source_id, source_capacity_in_bytes,
+                                            minimum_volume_size_in_bytes):
+        source_name = self._get_volume_name_by_wwn(source_id)
+        target_volume_name = self._get_volume_name_by_wwn(volume_id)
+        self._copy_to_target_volume(target_volume_name, source_name)
```

Comment:
now that we get the source *id* (instead of name) as a parameter, could you please change `_copy_to_target_volume` (and `_create_and_start_fcmap`) to require the source *id* accordingly?

Reply:
in this stage its not gonna save any code